server/controllers/users/helpers/validateUser.js

```diff
@@ -0,0 +1,18 @@
+const Joi = require('@hapi/joi');
+
+const schema = Joi.object().keys({
+  user_name: Joi.string()
+    .alphanum()
+    .min(3)
+    .required(),
+  full_name: Joi.string().min(6),
+  role: Joi.string(),
+  password: Joi.string().min(6),
+});
+
+module.exports = (req, res, next) => {
+  const user = req.body;
+  Joi.validate(user, schema)
+    .then(() => next())
+    .catch(() => next({ code: 401 }));
```

Comment:
`401` as a status code means an unauthorized request
https://httpstatuses.com/401
maybe we should use 422? try to read about it, I am not really sure, ask Farah, AbdSamdad and Ali about their opinion on this.

**Also,** check [this](https://blog.restcase.com/rest-api-error-codes-101/)  and let me know your opinion on it.